src/tokens.ts

```diff
@@ -236,46 +236,50 @@ export class TokenComputer {
     extractNonceFromExtendedIdentifier(identifier: string): number {
         const parts = identifier.split("-");
 
-        this.checkIfExtendedIdentifierWasProvided(parts);
-        this.checkLengthOfRandomSequence(parts[1]);
+        const { prefix, ticker, randomSequence } = this.splitIdentifierIntoComponents(identifier);
+        this.validateExtendedIdentifier(prefix, randomSequence, ticker, parts);
 
-        // in case the identifier of a fungible token is provided
-        if (parts.length == 2) {
+        // If identifier is for a fungible token (2 parts or 3 with prefix), return 0
+        if (parts.length === 2 || (prefix && parts.length === 3)) {
             return 0;
         }
 
-        const hexNonce = Buffer.from(parts[2], "hex");
-        return decodeUnsignedNumber(hexNonce);
+        // Otherwise, decode the last part as an unsigned number
+        const hecNonce = parts[parts.length - 1];
+        return decodeUnsignedNumber(Buffer.from(hecNonce, "hex"));
     }
 
     extractIdentifierFromExtendedIdentifier(identifier: string): string {
         const parts = identifier.split("-");
-        const { prefix, ticker, randomSequence } = this.splitIdentifierIntoComponents(parts);
+        const { prefix, ticker, randomSequence } = this.splitIdentifierIntoComponents(identifier);
 
-        this.checkIfExtendedIdentifierWasProvided(parts);
-        this.ensureTokenTickerValidity(ticker);
-        this.checkLengthOfRandomSequence(randomSequence);
+        this.validateExtendedIdentifier(prefix, randomSequence, ticker, parts);
         if (prefix) {
             this.checkLengthOfPrefix(prefix);
             return prefix + "-" + ticker + "-" + randomSequence;
         }
         return ticker + "-" + randomSequence;
     }
 
+    private validateExtendedIdentifier(prefix: string, randomSequence: string, ticker: string, parts: string[]): void {
```

Comment:
prefix can be `string | null`

Reply:
also reorder args to represent the token structure (prefix, ticker, random sequence)